### DISCUSSION TITLE
Target compilation question

### DISCUSSION THREAD

**Discussion by mscofield0** [mesonbuild/meson#9620] [Q&A] [answered]
How do I prevent target B being compiled when compiling target A, when both target A and B depend on target C?

The problem is that when I try compiling my `test` target via `meson test` or `ninja test`, the `app` target also gets compiled.

**Comment by eli-schwartz** [comment, marked as answer]
https://github.com/mesonbuild/meson/issues/1704

Basically, this is by design. But if you run specific tests using `meson test testname` it will only rebuild the explicit dependencies of that one test program -- or `meson test --no-rebuild` will assume you've taken care of building all test dependencies and not do so itself.

**Reply by mscofield0** [reply]
Ah, okay. Thanks.
